memberships/tests/test_stripe_webhooks.py

```diff
@@ -114,3 +114,26 @@ def test_a_successful_payment_for_membership_gets_logged_in_db(self):
         payments = Payment.objects.all()
 
         self.assertEqual(1, payments.count())
+
+    def test_member_gets_paid_permission_upon_successful_payment(self):
+        Membership.objects.create(
+            member=self.member, stripe_subscription_id=self.member.email
+        )
+        response = self.client.post(
+            reverse("stripe_webhook"),
+            {
+                "type": "invoice.payment_succeeded",
+                "data": {
+                    "object": {
+                        "customer_email": "test@example.com",
+                        "subscription": "sub_12345",
+                    }
+                },
+                "created": 1611620481,
+            },
+            content_type="application/json",
+        )
+        perm = Permission.objects.get(codename="has_sand_membership")
+        user = User.objects.get(id=self.member.user_id)
+
+        self.assertEqual(True, user.has_perm("has_sand_membership"))
```

Comment:
@CarwynNelson This returns `False` as the user permissions does not seem to update.
I have read that this is to do with the way that Django stores them in cache so the user needs to be reloaded?

I have tried defining the user again, something like `user = User.objects.get(pk=user.pk)` in the test and `views.py` to update the user, but this did not work.

What would you recommend?

https://stackoverflow.com/questions/27927159/django-1-7-adding-permissions-programmatically-does-not-work-as-expected

Reply:
@SamWinterhalder have you had any more luck on this?

Reply:
@CarwynNelson Haven't had a chance to investigate yet, will do either later today or tomorrow

Reply:
No worries, there is no rush :)

Reply:
I have had no luck. I am open to suggestions or am happy for someone else to take over.

Reply:
I was missing a `memberships.` before using the permission name... 18 DAYS